cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/KafkaPartitionMetricSampleOngoingExecutionStore.java

```diff
@@ -0,0 +1,108 @@
+
+  protected static final String PRODUCER_CLIENT_ID = "KafkaCruiseControlPartitionMetricSampleOnExecutionStoreProducer";
+
+  protected String _partitionMetricSampleStoreTopic;
+  protected int _partitionSampleStoreTopicPartitionCount;
+  protected long _partitionSampleStoreTopicRetentionTimeMs;
+
+  public static final String PARTITION_METRIC_SAMPLE_STORE_ON_EXECUTION_TOPIC_CONFIG = "partition.metric.sample.store.on.execution.topic";
+  public static final String PARTITION_METRIC_SAMPLE_STORE_ON_EXECUTION_TOPIC_REPLICATION_FACTOR_CONFIG =
+      "partition.metric.sample.store.on.execution.topic.replication.factor";
+  public static final String PARTITION_METRIC_SAMPLE_STORE_ON_EXECUTION_TOPIC_PARTITION_COUNT_CONFIG =
+      "partition.metric.sample.store.on.execution.topic.partition.count";
+  public static final String PARTITION_METRIC_SAMPLE_STORE_ON_EXECUTION_TOPIC_RETENTION_TIME_MS_CONFIG =
+      "partition.metric.sample.store.on.execution.topic.retention.time.ms";
+
+  @Override
+  public void configure(Map<String, ?> config) {
+    _partitionMetricSampleStoreTopic = KafkaCruiseControlUtils.getRequiredConfig(config,
+                                                                                 PARTITION_METRIC_SAMPLE_STORE_ON_EXECUTION_TOPIC_CONFIG);
+    String metricSampleStoreTopicReplicationFactorString = (String) config.get(
+        PARTITION_METRIC_SAMPLE_STORE_ON_EXECUTION_TOPIC_REPLICATION_FACTOR_CONFIG);
+    _sampleStoreTopicReplicationFactor = metricSampleStoreTopicReplicationFactorString == null
+                                         || metricSampleStoreTopicReplicationFactorString.isEmpty()
+                                         ? null : Short.parseShort(metricSampleStoreTopicReplicationFactorString);
+    String partitionSampleStoreTopicPartitionCountString = (String) config.get(
+        PARTITION_METRIC_SAMPLE_STORE_ON_EXECUTION_TOPIC_PARTITION_COUNT_CONFIG);
+    _partitionSampleStoreTopicPartitionCount = partitionSampleStoreTopicPartitionCountString == null
+                                               || partitionSampleStoreTopicPartitionCountString.isEmpty()
+                                               ? DEFAULT_PARTITION_SAMPLE_STORE_TOPIC_PARTITION_COUNT
+                                               : Integer.parseInt(partitionSampleStoreTopicPartitionCountString);
+    String sampleStoreTopicRetentionTimeMsString = (String) config.get(
+        PARTITION_METRIC_SAMPLE_STORE_ON_EXECUTION_TOPIC_RETENTION_TIME_MS_CONFIG);
+    _partitionSampleStoreTopicRetentionTimeMs = sampleStoreTopicRetentionTimeMsString == null
+                                                || sampleStoreTopicRetentionTimeMsString.isEmpty()
+                                                ? DEFAULT_PARTITION_SAMPLE_STORE_TOPIC_RETENTION_TIME_MS
+                                                : Long.parseLong(sampleStoreTopicRetentionTimeMsString);
```

Comment:
Looks like (1) `_partitionSampleStoreTopicPartitionCount` and (2) `_partitionSampleStoreTopicRetentionTimeMs` are used only once to ensure that the topic is created. So rather than keeping variables throughout the lifetime of the `KafkaPartitionMetricSampleOngoingExecutionStore` object, maybe we can just extract their values from config in `ensureTopicsCreated(Map<String, ?> config)` and pass them to `wrapTopic`.

-- Note: the same applies to `KafkaSampleStore`, ideally we should refactor it as well, but feel free to ignore this for that class as it was not introduced in this PR.